pywbem_mock/_dmtf_cim_schema.py

```diff
@@ -0,0 +1,426 @@
+environment like github. Therefore, a method
+:meth:`~pywbem_mock.DMTFCIMSchema.clean` removes all of the MOF files from
+local storage.  They are restored the next time the
+:class:`~pywbem_mock.DMTFCIMSchema` object for the same version is constructed.
+
+The DMTF maintains two versions of each released schema:
+
+* Final - This schema does not contain any of the schema classes that are
+  marked experimantel.  It is considered the stable release.
+
+* Experimental - Contains the classes in the `Final` schema file plus other
+  classes that are considered experimental. Typically it is considered less
+  stable.
+
+:class:`~pywbem_mock.DMTFCIMSchema` can install either the `Final` or
+'experimental' schema depending on the value of the `user-experimental`
+parameter.
+
+The :class:`~pywbem_mock.DMTFCIMSchema` class also provides a
+:meth:`~pywbem_mock.build_schema_mof` to create a MOF include file from the
+DMTF Schema downloaded that includes only a subset of the classes actually in
+the repository for compile into a MOF repository.
+"""
+import os
+from zipfile import ZipFile
+import shutil
+import six
+
+if six.PY2:
+    # pylint: disable=wrong-import-order
+    from urllib2 import urlopen
+else:
+    # pylint: disable=wrong-import-order
+    from urllib.request import urlopen
+
+
+__all__ = ['DMTFCIMSchema']
+
+
+class DMTFCIMSchema(object):
+    """
+    :class:`DMTFCIMSchema` represents a DMTF CIM Schema downloaded from the
+    DMTF WEB site.
+
+    This class manages the download and install of DMTF CIM schema releases into
+    a form that can be used by :class:`pywbem_mock.FakedWBEMConnection` to
+    create a repository with class and qualifier declarations.
+    """
+    def __init__(self, schema_version, schema_root_dir, use_experimental=False,
+                 verbose=False):
+        """
+        The constructor downloads the schema if the `schema_root_dir` or the
+        schema zip file do not exist and installs it into the directory defined
+        by the `schema_root_dir` parameter.
+
+        If the schema zip file is already installed it simply sets the property
+        values so that the schema can be compiled.
+
+        If the schema zip file was previously downloaded but not expanded, it is
+        expanded (i.e. the mof... subdirectory is created and the MOF files
+        expanded into this directory.).
+
+        Parameters:
+
+          schema_version  (tuple of 3 integers (x, y, z):
+            Represents the DMTF CIM schema version m.n.u where:
+
+            * m is the DMTF CIM schema major version
+            * n is the DMTF CIM schema minor version
+            * u is the DMTF CIM schema update version
+
+            This MUST represent a DMTF CIM schema that is available from the
+            DMTF web site.
+
+          schema_root_dir (:term:`string`):
+            Directory into which the DMTF CIM schema is installed or will be
+            installed.
```

Comment:
Several comments:
* Don't use "install", see comment above.
* Not clear whether this directory holds just the zip file (yes) or is also the root directory for the MOF tree (no).
* It is the path name of the directory, not the directory.

Suggestion to address all of that, replaciong the current sentence:

"Path name of the schema root directory into which the DMTF CIM schema zip file is downloaded and in which a subdirectory for the extracted MOF files is created.
+            installed.

Reply:
DONE